lib/order.rb

```diff
@@ -0,0 +1,32 @@
+class Order
+  attr_reader :selected_items
+  def initialize(menu_class = Menu, dish_class = Dish, confirmation_class = Confirmation)
+    @selected_items = []
+    @menu = menu_class.new([dish_class.new("chips", 1.50), dish_class.new("nuggets", 2.00),
+    dish_class.new("big mac", 3.50), dish_class.new("apple pie", 0.75)])
+    @total = 0
```

Comment:
Considering the total is derived from the dishes in the order, it's better to calculate it when needed, rather than store it in an instance variable. Duplicating state like this can cause bugs if they get out of sync, so you should try to stick to a single source of truth.